bootstrap/v2/pkg/kfapp/kustomize/kustomize.go

```diff
@@ -577,18 +578,55 @@ func (kustomize *kustomize) writeConfigFile() error {
 	return nil
 }
 
+func (kustomize *kustomize) overlayResources(compName string, compPath string) map[string][]string {
+	params := kustomize.Spec.ComponentParams[compName]
+	overlays := []string{}
+	if params != nil {
+		for _, nv := range params {
+			name := nv.Name
+			if name == "overlay" {
+				overlays = append(overlays, nv.Value)
+			}
+		}
+	}
+	resources := func (overlayPath string) []string {
+		resources := make([]string, 0)
+		files, err := ioutil.ReadDir(overlayPath)
+		if err != nil {
+			return resources
+		}
+		for _, f := range files {
+			if !f.IsDir() && f.Name() != "kustomization.yaml" && strings.HasSuffix(f.Name(), ".yaml") {
+				fpath := filepath.Join(overlayPath, f.Name())
+				resources = append(resources, extractSuffix(kustomize.Spec.ManifestsRepo, fpath))
+			}
+		}
+		return resources
+	}
+	overlayResourcesMap := make(map[string][]string)
+	for _, overlay := range overlays {
+		compAbsolutePath := path.Join(kustomize.Spec.ManifestsRepo, compPath)
+		overlayPath := path.Join(compAbsolutePath, overlay)
+		overlayResourcesMap[overlay] = resources(overlayPath)
+	}
+	return overlayResourcesMap
+}
+
 // writeKustomizationFile will write out a top-level kustomization.yaml under the manifests downloaded cache.
 // this file is then used to generate a component under <deployment>/kustomize
-func (kustomize *kustomize) writeKustomizationFile(compPath string) error {
+func (kustomize *kustomize) writeKustomizationFile(compName string, compPath string) error {
 	bases := []string{}
 	comp := path.Join(kustomize.Spec.ManifestsRepo, compPath)
 	compKustomizationFile := filepath.Join(comp, kftypes.KustomizationFile)
 	base := path.Join(kustomize.Spec.ManifestsRepo, compPath, "base")
 	baseKustomizationFile := filepath.Join(base, kftypes.KustomizationFile)
-	platformOverlay := path.Join(kustomize.Spec.ManifestsRepo, compPath, "overlays", kustomize.Spec.Platform)
-	platformKustomizationFile := filepath.Join(platformOverlay, kftypes.KustomizationFile)
-	if _, err := os.Stat(platformKustomizationFile); err == nil {
-		bases = append(bases, extractSuffix(kustomize.Spec.ManifestsRepo, platformOverlay))
+	overlays := kustomize.overlayResources(compName, compPath)
+	if kustomize.Spec.Platform != "" && len(overlays) == 0 {
```

Comment:
I think we should not ignore kustomize.Spec.Platform != ""  && len(overlays) >0 case.
for example, for minikube platform, by `overlays`, we can support if enable-gang-scheduling enabled or not, refer to [here](https://github.com/kubeflow/manifests/issues/27)